### DISCUSSION TITLE
Support extending views, not just models

### DISCUSSION THREAD

**Discussion by kcaswick** [likec4/likec4#313] [Ideas]
This would allow making extensions completely independent of the base files, which might also be maintained elsewhere. (i.e. a submodule from another team, an open source project that includes diagrams you want to reference, or an automatically generated file)

For example, in [next.c4](/likec4/likec4/tree/main/examples/cloud-system/likec4/cloud/next.c4), it could contain:

```likec4
views {
  extend cloud {
    include next.*
    exclude next -> legacy
  }
}
```

This would add the planned new system to views, without mentioning `next` at all in [views.c4](/likec4/likec4/tree/main/examples/cloud-system/likec4/views.c4), which might still be needed for projects that only need to see the legacy system.

It probably makes sense to allow extending in `specification` at the same time, though at the moment it would only be useful for having a centralized place to add styles to elements.

**Comment by davydkov** [comment]
Hi @kcaswick 

I already have a draft implementation (as a part of the "presentation/slides" mode I was working a while back), like

```
views {

  view index {
    include ...
  }
  
  view index2 extends index {  
    exclude ...
    include ..
    style ...
  }
  
  view index3 extends index2 {
    ...
  }

}
```

Implementation could be more subtle, but it does the job. I will release it soon and let you know.

Does this syntax look good for you?

**Reply by kcaswick** [reply]
Yeah, that looks pretty close. 

The only thing missing is being able to add a "viewOf", which I did when I was doing the workaround of copying it and then editing. If that's not easy, I can get by without it though. (I'm combining diagrams for 2 packages, so I want to move an index view from 1 package to be seen when clicking the package element for that package.)

Thanks!

**Reply by davydkov** [reply]
Released in [v0.37.0](https://github.com/likec4/likec4/releases/tag/v0.37.0).

As for “viewOf” - I want to have custom navigation rules, maybe defined per view.

Be honest, I have no idea yet how this might be presented in syntax.
